src/com/walmartlabs/lacinia/pedestal/subscriptions.clj

```diff
@@ -0,0 +1,403 @@
+(ns com.walmartlabs.lacinia.pedestal.subscriptions
+  "Support for GraphQL subscriptions using Jetty WebSockets, following the design
+  of the Apollo client and server."
+  {:added "0.3.0"}
+  (:require
+    [com.walmartlabs.lacinia.util :as util]
+    [clojure.core.async :as async
+     :refer [chan put! close! go-loop <! >! alt! thread]]
+    [cheshire.core :as cheshire]
+    [io.pedestal.interceptor :refer [interceptor]]
+    [io.pedestal.interceptor.chain :as chain]
+    [io.pedestal.http.jetty.websockets :as ws]
+    [io.pedestal.log :as log]
+    [com.walmartlabs.lacinia.parser :as parser]
+    [com.walmartlabs.lacinia.validator :as validator]
+    [com.walmartlabs.lacinia.executor :as executor]
+    [com.walmartlabs.lacinia.constants :as constants]
+    [com.walmartlabs.lacinia.resolve :as resolve]
+    [com.walmartlabs.lacinia.pedestal.interceptors :as interceptors
+     :refer [ordered-after]]
+    [clojure.string :as str]))
+
+(defn ^:private xform-channel
+  [input-ch output-ch xf]
+  (go-loop []
+    (if-some [input (<! input-ch)]
+      (let [output (xf input)]
+        (when (>! output-ch output)
+          (recur)))
+      (close! output-ch))))
+
+(defn ^:private response-encode-loop
+  "Takes values from the input channel, encodes them as a JSON string, and
+  puts them into the output-ch."
+  [input-ch output-ch]
+  (xform-channel input-ch output-ch cheshire/generate-string))
```

Comment:
I was not very happy with Pedestal's support here; I'm having to work around their use case to get to ours.  I may PR Pedestal with some small refactoring to make our approach easier.